daemon/prompter.c

```diff
@@ -274,6 +300,27 @@ prompter_delete_cb(void *self)
     prompter_delete(self);
 }
 
+void
+prompter_applications_changed(prompter_t *self, const stringset_t *changed)
+{
+    /* First check the current invocation */
+    if( prompter_check_if_changed(self, prompter_current_invocation(self),
+                                  changed) ) {
+        change_cancellable(&self->prm_cancellable, NULL);
```

Comment:
I don't like the name either. I'll try to come up with a better one.

Reply:
I changed it to `prompter_try_finish_invocation()` although it's not exactly the same as `prompter_finish_invocation()` with the possibility of not changing anything as it may apply also to other invocations than the current one.

Reordered the functions too.

Reply:
prompter_try_finish_invocation() = makes intent clear enough